security/pkg/nodeagent/sds/server.go

```diff
@@ -169,6 +170,7 @@ func (s *Server) initWorkloadSdsService(options *Options) error {
 			log.Info("Start SDS grpc server")
 			if err = s.grpcWorkloadServer.Serve(s.grpcWorkloadListener); err != nil {
 				log.Errorf("SDS grpc server for workload proxies failed to start: %v", err)
+				time.Sleep(1 * time.Second) // without this the logs fill up with thousands of the message
```

Comment:
which message to the logs fill with?

Reply:
The message is 

```
2019-02-20T19:54:28.929717Z     info    Start SDS grpc server
2019-02-20T19:54:28.929737Z     error   SDS grpc server for workload proxies failed to start: grpc: the server has been stopped
```

Running an integration `TestStreamSecretsForWorkloadSds` I saw the message repeat 16198 times during 2.5 seconds.

Reply:
Is it a bug then?

Reply:
For me the repeated message overwhelms test failure output.  I assume if there was a failure in product it will challenge the logging infrastructure.  I don't know about the root cause or what the node agent should do in this situation.

Reply:
It seems strange that this method is being called repeatedly upon error .... I think we need to understand why that's happening.  Adding this sleep is just masking the issue.

Reply:
I have removed the sleep.  Anything else needed before this can be merged?  It will help with https://github.com/istio/istio/issues/11593 .